barcode_generator_product_variant/wizards/product_product_barcode_wizard.py

```diff
@@ -0,0 +1,75 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class ProductProductBarcodeWizard(models.TransientModel):
+    _name = "product.product.barcode.wizard"
+    _description = "Wizard to manage barcode generation "
+    "for variants of a product"
+
+    def _default_barcode_rule_id(self):
+        return self.env["barcode.rule"].search(
+            [("is_default_rule", "=", True)], limit=1
+        )
+
+    product_tmpl_id = fields.Many2one(
+        "product.template", required=True, readonly=True, string="Product"
+    )
+    product_ids = fields.Many2many(
+        "product.product",
+        compute="_compute_product_ids",
+        store=True,
+        domain="[('product_tmpl_id', '=', product_tmpl_id)]",
+        string="Variants",
+    )
+    barcode_rule_id = fields.Many2one(
+        string="Barcode Rule",
+        comodel_name="barcode.rule",
+        default=_default_barcode_rule_id,
+        domain="""[
+            ('generate_model', '=', 'product.product'),
+            ('generate_type', '=', 'sequence'),
+            ]""",
+    )
+
+    def name_get(self):
+        return [
+            (
+                rec.id,
+                _(
+                    "Barcode management for %(prod)s",
+                    prod=rec.product_tmpl_id.display_name,
+                ),
+            )
+            for rec in self
+        ]
+
+    @api.depends("product_tmpl_id")
+    def _compute_product_ids(self):
+        for wiz in self:
+            if not wiz.product_tmpl_id:
+                wiz.product_ids = False
+                continue
+            wiz.product_ids = wiz.product_tmpl_id.product_variant_ids.filtered(
+                lambda x: not x.barcode
+            )
```

Comment:
nitpicking : simplification.

```suggestion
        for wizard in self:
            wizard.product_ids = wizard.mapped("product_tmpl_id.product_variant_ids").filtered(
                lambda x: not x.barcode
            )
```